tools/CheckPowershellCompliance.ps1

```diff
@@ -0,0 +1,139 @@
+# This Powershell script checks Powershell compatibility of any custom scripts
+# in the repository.  It assumes configuration for this is present in the
+# README.md. 
+#
+# The configuration should look like this:
+#
+#     ### PSScriptAnalyzer Configuration
+#     ``` yaml
+#     targetVersions:
+#     - "5.1"
+#     - "7.0"
+#     targetProfiles:
+#     - "win-8_x64_10.0.17763.0_5.1.17763.316_x64_4.0.30319.42000_framework"
+#     - "win-8_x64_10.0.14393.0_7.0.0_x64_3.1.2_core"
+#     ```
+
+function Read-ComplianceConfiguration {
+    $path = "README.md"
+    $configMarker = "### PSScriptAnalyzer Configuration"
+    $backTicksYaml = '``` yaml'
+    $backTicks = '```'
+
+    $readme = Get-Content $path
+
+    # Extract out the YAML which sits between backticks markers after the 
+    # PSScriptAnalyzer configuration marker.
+    $yaml = ""
+    $inConfig = $false
+    for ($i = 0; $i -lt $readme.Length; $i++) {
+        if ($readme[$i] -eq $configMarker) {
+            $inConfig = $true
+        }
+        if ($inConfig -and ($readme[$i] -eq $backTicksYaml)) {
+            $i++
+            while ($i -lt $readme.Length -and $readme[$i] -ne $backticks) {
+                $yaml += "`n" + $readme[$i]
+                $i++
+            }
+            break
+        }
+    }
+
+    if ("" -eq $yaml) {
+        Write-Error "No PSScriptAnalyzer Configuration found in README.md"
```

Comment:
I notice in the script **write-error** is widely used. I think **Write-Host** should be used for most cases. Some services may not have configurations for it. We could just say something like 'No configuration found. Skipped'. Or we could have a default configuration set in this script. If no configuration is specified in README.md, we just use the default one.

Reply:
I thought that `Write-Host` was deprecated and the `PSScriptAnalyzer` complains about it.
As to whether no configuration is an error or not, happy to change this is there becomes a way to **always** run this script but at present it has to be run manually so it seems reasonable to report an error along the lines of "you expressly thought you were going to be running this check but you didn't configure it!".

Reply:
I see. One more comment.
`write-error` is not a terminate error by default, which means the script will keep executing.

Please use `throw "some message"` instead of `write-error 'some message';  exit` for a terminate error.